packages/components/src/box-control/input-control.tsx

```diff
@@ -0,0 +1,211 @@
+/**
+ * WordPress dependencies
+ */
+import { useInstanceId } from '@wordpress/compose';
+import { __ } from '@wordpress/i18n';
+
+/**
+ * Internal dependencies
+ */
+import Tooltip from '../tooltip';
+import { parseQuantityAndUnitFromRawValue } from '../unit-control/utils';
+import {
+	ALL_SIDES,
+	CUSTOM_VALUE_SETTINGS,
+	getAllValue,
+	isValuesDefined,
+	isValuesMixed,
+	LABELS,
+} from './utils';
+import {
+	FlexedBoxControlIcon,
+	FlexedRangeControl,
+	InputWrapper,
+	StyledUnitControl,
+} from './styles/box-control-styles';
+import type { BoxControlInputControlProps, BoxControlValue } from './types';
+
+const noop = () => {};
+
+function getAllowedSides( sides: BoxControlInputControlProps[ 'sides' ] ) {
+	const allowedSides: Set< keyof BoxControlValue > = new Set(
+		! sides ? ALL_SIDES : []
+	);
+	sides?.forEach( ( allowedSide ) => {
+		if ( allowedSide === 'vertical' ) {
+			allowedSides.add( 'top' );
+			allowedSides.add( 'bottom' );
```

Comment:
We might need to reorder these.

Reply:
How so?

Reply:
Actually, I confirm it's not necessary after reading the code again.